src/qml/imports/QFieldControls/+Qt5/CodeReader.qml

```diff
@@ -29,50 +31,88 @@ Popup {
 
   onAboutToShow: {
     openedOnce = true
+    // when NFC is not accessible, make sure the only option, QR, is active
+    if (!withNfc && !settings.cameraActive) {
+      settings.cameraActive = true
+    }
     barcodeDecoder.clearDecodedString();
   }
 
   onAboutToHide: {
-    cameraLoader.item.flash.mode = Camera.FlashOff;
+    if (cameraLoader.active) {
+      cameraLoader.item.flash.mode = Camera.FlashOff;
+    }
+  }
+
+  Settings {
+    id: settings
+    property bool cameraActive: true
+    property bool nearfieldActive: true
   }
 
   BarcodeDecoder {
     id: barcodeDecoder
 
     onDecodedStringChanged: {
       if (decodedString !== '') {
+        codeReader.decodedString = decodedString
         decodedFlashAnimation.start();
       }
     }
   }
 
   Loader {
-    id: cameraLoader
-    sourceComponent: cameraComponent
-    active: barcodeReader.openedOnce
+    id: nearfieldLoader
+    active: withNfc && codeReader.openedOnce && settings.nearfieldActive
+
+    sourceComponent: Component {
+      Item {
+        id: nearFieldContainer
+
+        Component.onCompleted: {
+          Qt.createQmlObject('import org.qfield 1.0
+            NearFieldReader {
+              active: codeReader.visible
+              onTargetDetected: (targetId) => {
+                displayToast(qsTr(\'NFC tag detected\'))
```

Comment:
Can we make it somehow visible to a user when an nfc tag without text has been detected? I.e. in the camera screen I find this message looks like "success" even if there's no text captured.

Reply:
Pretty sure this is not picked up for translation either.

Reply:
Fair point. Reason why this is done is explained above, but all the same, seems like we'd be better of a dummy NearFieldReader instead of none whatsoever.